### PR TITLE
WIP - API Docs V2: redoc -> scalar + better docstrings

### PR DESCRIPTION
**Pull request by Theodlz**
Work in progress.

Essentially this moves away from redoc in favor of a much nicer tool (scalar), and also adds a few things such as a "summary" key to the docstrings of all the handlers, so that a readable text shows up in the list of endpoints rater than an obscure endpoint regex.

The reviewer can just run `make api-docs` to generate it locally and then can open `openapi.html` to see them

### CODE REVIEW

**Comment by Theodlz** [comment]
@stefanv I've deactivated the "try it" feature for now, until we fix the CORS. It's a nice feature but it's not critical, so I'd suggest doing that in a future PR.

File: tools/docs/patch-template.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+with open('openapi.json') as f:
+    openapi_spec = json.load(f)
+
+# here, we could add a list of servers to the openapi_spec
+# for the scalar docs to let the users point to when testing an endpoint
+servers = cfg.get("docs.servers", [])
+# validate that servers have only 2 keys: url and description
+if not isinstance(servers, list) and servers is not None:
+    raise ValueError("API servers must be a list.")
+if servers is not None:
```
Comment (stefanv) [inline tools/docs/patch-template.py:20]: Two checks that servers is not none; maybe handle that a different way on line 16

File: tools/docs/patch-template.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    for server in servers:
+        if not all(k in server for k in ("url", "description")):
+            raise ValueError("Each server must have 'url' and 'description' keys.")
```
Comment (stefanv) [inline tools/docs/patch-template.py:23]:
```suggestion
    if not all(key in server for key in ("url", "desc") for server in servers)
        raise ValueError("Each server must have 'url' and 'description' keys.")
```

File: Makefile [hunk 3]
```diff
@@ -36,13 +36,9 @@ doc_reqs:
 
 api-docs: FLAGS := $(if $(FLAGS),$(FLAGS),--config=config.yaml)
 api-docs: | doc_reqs
-	@PYTHONPATH=. python tools/openapi/build-spec.py $(FLAGS)
-	npx @redocly/cli build-docs openapi.json \
-          --title "SkyPortal API docs" \
-          --theme.openapi.logo.gutter 2rem
+	@PYTHONPATH=. python tools/docs/build-spec.py $(FLAGS)
+	@PYTHONPATH=. python tools/docs/patch-template.py $(FLAGS)
```
Comment (stefanv) [inline Makefile:40]: Perhaps a more descriptive name? Something like `patch-api-doc-template.py`. And we could rename the other one to `build-api-spec.py`.

**Comment by stefanv** [comment]
image

Some font size weirdness :/

**Review by stefanv — APPROVED** [review]

**Review by mcoughlin — APPROVED** [review]
Dis coooolll

File: tools/docs/patch-template.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import json
```
[the remark this answers is not in the document]
Comment (Theodlz) [reply tools/docs/patch-template.py]: done, made it a function and added a docstring

File: tools/docs/patch-template.py (hunk 1 above)
Comment (Theodlz) [reply tools/docs/patch-template.py]: done

File: Makefile (hunk 3 above)
Comment (Theodlz) [reply Makefile]: done

**Comment by Theodlz** [comment]
addressed all of your comments I believe @stefanv

**Comment by Theodlz** [comment]
> image
> 
> Some font size weirdness :/ (Minor.)
> 
> Otherwise, looks great!

yeah... I'll open an issue on scalar's repo. I opened 2 yesterday and they responded super quickly (they are super active, it's awesome and very reassuring).

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Good stuff, thanks Theo!

**Pull request merged**